pandas/io/parsers.py

```diff
@@ -86,13 +87,14 @@
     MultiIndex is used. If you have a malformed file with delimiters at the end
     of each line, you might consider index_col=False to force pandas to _not_
     use the first column as the index (row names)
-usecols : array-like, default None
-    Return a subset of the columns. All elements in this array must either
+usecols : array-like or callable, default None
+    Return a subset of the columns. If array-like, all elements must either
     be positional (i.e. integer indices into the document columns) or strings
     that correspond to column names provided either by the user in `names` or
     inferred from the document header row(s). For example, a valid `usecols`
-    parameter would be [0, 1, 2] or ['foo', 'bar', 'baz']. Using this parameter
-    results in much faster parsing time and lower memory usage.
+    parameter would be [0, 1, 2], ['foo', 'bar', 'baz'] or lambda x: x.upper()
+    in ['AAA', 'BBB', 'DDD']. Using this parameter results in much faster
+    parsing time and lower memory usage.
```

Comment:
Can you add before the example some explanation how the callable is used? (I suppose it is called on each individual column name and should return True or False?)

And maybe also use ``` `` ``` around the code example to better indicate that the "in ['AAA', 'BBB', 'DDD']" still belongs with the "lambda x: x.upper()"